File: src/main/java/bbarena/model/util/Concat.java
package bbarena.model.util;

public class Concat {
	
	public static String buildLog(Class<?> type, Pair... pairs) {
		StringBuilder res = new StringBuilder();
		res.append(type.getSimpleName())
		.append("[");
		if (pairs != null) {
			for (int i = 0; i < pairs.length; i++) {
				if (pairs[i] != null) {
					res.append(pairs[i].toString());
					if (i < pairs.length - 1) {
						res.append(",");
					}
				}
			}
		}
		res.append("]");
		return res.toString();
	}
	
}

File: src/main/kotlin/bbarena/model/util/Concat.kt
package bbarena.model.util

object Concat {

    fun buildLog(type: Class<*>, vararg pairs: Pair<Any, Any>): String {
        val res = StringBuilder()
        res.append(type.simpleName)
                .append(pairs.joinToString(separator = ";", prefix = "[", postfix = "]") { it.toString() })
        return res.toString()
    }

}
